src/ipc/ipc4/handler.c

```diff
@@ -1063,6 +1063,16 @@ void ipc_boot_complete_msg(struct ipc_cmd_hdr *header, uint32_t data)
 	header->ext = 0;
 }
 
+void ipc_send_panic_notification(void)
+{
+	msg_notify.header = SOF_IPC4_NOTIF_HEADER(SOF_IPC4_EXCEPTION_CAUGHT);
+	msg_notify.extension = cpu_get_id();
+	msg_notify.tx_size = 0;
+	list_init(&msg_notify.list);
+
+	ipc_msg_send_direct(&msg_notify, NULL);
```

Comment:
I see that currently `msg_notify.tx_data` is never set, but if it's ever set to `!= NULL`, then the `msg->tx_data != data` inside `ipc_msg_send_direct()` (and `ipc_msg_send()`) will return `true` and they will try to copy from `NULL`. It's not a very robust API...

Reply:
@lyakh  I set msg_notify.tx_data to NULL. Thanks!  Please comment if you have more ideas about it.